.hooks/pre-commit

```diff
@@ -4,4 +4,27 @@ set -e
 
 # consider using pre-push instead
 
+# get all staged files
+staged_files=$(git diff --cached --name-only)
+
+# if **/Data/*.php or **/Enums/*.php is among staged files, sync types
+if echo "$staged_files" | grep -E "(Data|Enums)/.+\.php$"; then
+    printf "\n🔄 Changes detected in back-end DTOs."
+    printf "\n⏳ composer types\n\n"
+    composer types
+
+    # staged all generated TypeScript files
+    git add resources/js/types/generated.d.ts resources/js/common/utils/generatedAppConstants.ts
```

Comment:
I have concerns about forcing this. I've noticed that running the command on my machine generates things in a different order than what's committed. It could lead to a series of alternating commits between users that are effectively no-ops.
```
jamiras@ubuntu24 master:~/source/RAWeb$ git status
On branch master
Your branch is up to date with 'upstream/master'.

nothing to commit, working tree clean
jamiras@ubuntu24 master:~/source/RAWeb$ composer types
> @php artisan typescript:generate
TypeScript definitions and constants generated successfully.
> @php artisan ziggy:generate --types-only
Files generated!
jamiras@ubuntu24 master:~/source/RAWeb$ git status
On branch master
Your branch is up to date with 'upstream/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   resources/js/common/utils/generatedAppConstants.ts
	modified:   resources/js/ziggy.d.ts
jamiras@ubuntu24 master:~/source/RAWeb$ git diff | wc -l
575
```

Reply:
No worries. Forcing `composer types` may indeed be a little assertive right now. In latest, I've updated the hook so only the i18n is forced.